client/me/purchases/confirm-cancel-purchase/index.jsx

```diff
@@ -24,43 +24,36 @@ const ConfirmCancelPurchase = React.createClass( {
 		selectedSite: React.PropTypes.object
 	},
 
-	componentWillMount() {
-		recordPageView( 'confirm_cancel_purchase', this.props );
-	},
-
-	componentWillReceiveProps( nextProps ) {
-		recordPageView( 'confirm_cancel_purchase', this.props, nextProps );
+	getInitialState() {
+		return {
+			isFormLoaded: false
+		};
 	},
 
-	componentDidMount() {
-		if ( ! this.props.selectedPurchase.hasLoadedFromServer || ! this.props.selectedSite ) {
-			return;
-		}
-
-		this.resetEndpointForm();
-	},
-
-	componentDidUpdate( prevProps ) {
-		if ( ! this.props.selectedPurchase.hasLoadedFromServer ) {
-			return;
-		}
-
-		if ( isEqual( prevProps.selectedPurchase.data, this.props.selectedPurchase.data ) &&
-				( prevProps.selectedSite.ID === this.props.selectedSite.ID ) ) {
-			return;
-		}
-
-		this.resetEndpointForm();
+	componentWillMount() {
+		this.loadEndpointForm();
+		recordPageView( 'confirm_cancel_purchase', this.props );
 	},
 
-	resetEndpointForm() {
-		const { selectedSite, selectedPurchase } = this.props;
-
-		loadEndpointForm( {
-			container: React.findDOMNode( this.refs.root ),
-			selectedSite,
-			selectedPurchase: selectedPurchase.data,
-			onSubmit: this.handleSubmit
+	loadEndpointForm() {
+		const purchase = getPurchase( this.props );
+
+		loadEndpointForm( purchase, ( html, initializeForm ) => {
+			if ( ! this.isMounted() ) {
+				return;
+			}
+
+			this.setState( { isFormLoaded: true }, () => {
+				const container = React.findDOMNode( this.refs.root );
+
+				container.innerHTML = html;
```

Comment:
This isn't part of `render` method, so we can't really use `_dangerouslySetInnerHTML` here. In the previous version I coded it in a different way where `html` code was stored in `state`, but I had to use another lifecycle method to initialize form. @drewblaisdell made good point that current approach should work better:
> My rationale:
* This uses the minimal state to represent render().
* We only want to inject the form once, so this is a little more explicit than using componentDidUpdate with a probably-one-time condition like ! prevState.html && this.state.html.